wrapper/src/main/java/software/amazon/jdbc/Driver.java

```diff
@@ -139,13 +158,30 @@ public Connection connect(final String url, final Properties info) throws SQLExc
       PARENT_LOGGER.setLevel(logLevel);
     }
 
-    final TargetDriverDialectManager targetDriverDialectManager = new TargetDriverDialectManager();
-    final TargetDriverDialect targetDriverDialect =
-        targetDriverDialectManager.getDialect(driver, info);
+    TargetDriverDialect targetDriverDialect = configurationProfile == null
```

Comment:
Minor nitpick: It might be a bit easier to understand if the assignment of `targetDriverDialect` only done in the `if` block. 
For example
```
TargetDriverDialect targetDriverDialect;
if (configurationProfile == null) {
  final TargetDriverDialectManager targetDriverDialectManager = new TargetDriverDialectManager();
  targetDriverDialect = targetDriverDialectManager.getDialect(driver, props);
} else {
  targetDriverDialect = configurationProfile.getTargetDriverDialect();
}
```